datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeBackupRestoreStatusResponseConverter.java

```diff
@@ -11,31 +16,119 @@
+            } else {
+                failureString.append(getFailure(OperationEnum.HBASE_ATLAS_JANUS.description(), hbase.getAtlasJanusTable()));
+                failureString.append(getFailure(OperationEnum.HBASE_ATLAS_AUDIT.description(), hbase.getAtlasEntityAuditEventTable()));
+            }
+        }
+    }
+
+    private void parseSolrFailure(datalakeDRProto.SolrBackupRestoreState solr, StringBuilder failureString) {
+        if (solr != null) {
+            List<datalakeDRProto.BackupRestoreOperationStatus> allFailureReasons = List.of(
+                    solr.getEdgeIndexCollection(),
+                    solr.getFulltextIndexCollection(),
+                    solr.getRangerAuditsCollection(),
+                    solr.getVertexIndexCollection()
+            );
+            if (StringUtils.isNotEmpty(solr.getEdgeIndexCollection().getFailureReason()) && areAllFailuresTheSame(allFailureReasons)) {
+                failureString.append(getFailure(OperationEnum.SOLR.description(), solr.getEdgeIndexCollection()));
+            } else {
+                failureString.append(getFailure(OperationEnum.SOLR_EDGE_INDEX.description(), solr.getEdgeIndexCollection()));
+                failureString.append(getFailure(OperationEnum.SOLR_FULLTEXT_INDEX.description(), solr.getFulltextIndexCollection()));
+                failureString.append(getFailure(OperationEnum.SOLR_RANGER_AUDITS.description(), solr.getRangerAuditsCollection()));
+                failureString.append(getFailure(OperationEnum.SOLR_VERTEX_INDEX.description(), solr.getVertexIndexCollection()));
+            }
+            List<datalakeDRProto.BackupRestoreOperationStatus> allDeleteFailureReasons = List.of(
+                    solr.getEdgeIndexCollectionDelete(),
+                    solr.getFulltextIndexCollectionDelete(),
+                    solr.getRangerAuditsCollectionDelete(),
+                    solr.getVertexIndexCollectionDelete()
+            );
+            if (StringUtils.isNotEmpty(solr.getEdgeIndexCollectionDelete().getFailureReason()) && areAllFailuresTheSame(allDeleteFailureReasons)) {
+                failureString.append(getFailure(OperationEnum.SOLR_DELETE.description(), solr.getEdgeIndexCollection()));
+            } else {
+                failureString.append(getFailure(OperationEnum.SOLR_EDGE_INDEX_DELETE.description(), solr.getEdgeIndexCollectionDelete()));
+                failureString.append(getFailure(OperationEnum.SOLR_FULLTEXT_INDEX_DELETE.description(), solr.getFulltextIndexCollectionDelete()));
+                failureString.append(getFailure(OperationEnum.SOLR_RANGER_AUDITS_DELETE.description(), solr.getRangerAuditsCollectionDelete()));
+                failureString.append(getFailure(OperationEnum.SOLR_VERTEX_INDEX_DELETE.description(), solr.getVertexIndexCollectionDelete()));
+            }
+        }
+    }
+
+    private String getFailure(String operationName, datalakeDRProto.BackupRestoreOperationStatus status) {
```

Comment:
please use `Optional<String>` to avoid empty string usage

Reply:
and during append you can use this:
`getFailure(OperationEnum.SOLR_EDGE_INDEX_DELETE.description(), solr.getEdgeIndexCollectionDelete()).ifPresent(failureString::append);`

Reply:
Done.